### PR TITLE
Add API documentation

### PR DESCRIPTION
**Pull request by Carlgo11**
A more comprahensive documentation than the one that #5075 removes.

Unsure about the grammar. @jamcat22 might want to look at that 🙈

### CODE REVIEW

**Comment by phallobst** [comment]
Maybe it would better to put the v2 paragraph _before_ v1 to indicate the preference.

File: API.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+|---|--------|
+|https://twofactorauth.org/api/v2/all.json|All sites|
+|https://twofactorauth.org/api/v2/tfa.json|All 2FA-supporting sites|
+|https://twofactorauth.org/api/v2/sms.json|SMS|
+|https://twofactorauth.org/api/v2/phone.json|Phone calls|
+|https://twofactorauth.org/api/v2/email.json|Email 2FA|
+|https://twofactorauth.org/api/v2/hardware.json|non-U2F hardware 2FA tokens|
+|https://twofactorauth.org/api/v2/u2f.json|U2F hardware tokens|
+|https://twofactorauth.org/api/v2/totp.json|RFC-6238|
+|https://twofactorauth.org/api/v2/proprietary.json|non-RFC-6238 software 2FA
```
Comment (phallobst) [inline API.md]:
```suggestion
|https://twofactorauth.org/api/v2/proprietary.json|non-RFC-6238 software 2FA|
```

File: API.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+|Key|Value|Always defined|Description|
+|---|-----|---------------|-----------|
+|url|URL|:heavy_check_mark:|URL to the main page of the site/service|
+|img|String|:heavy_check_mark:|Image name used|
+|tfa|Boolean|:heavy_check_mark:|2FA support|
+|sms|Boolean||SMS token support|
+|phone|Boolean||Phone call support|
+|email|Boolean||Email token support|
+|software|Boolean||Software token support (including RFC-6238)|
+|hardware|Boolean||Hardware token support (including U2F-tokens)|
```
Comment (phallobst) [inline API.md]:
```suggestion
|hardware|Boolean||Hardware token support (including U2F tokens)|
```

File: API.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    }  
+  }
+}
+```
+
+## Notes and remarks
+
+### URL & Domain matching
+
+If you're using our API to match client URLs with our dataset make sure you're only use the domain of the `url`-element as a website commonly uses subdomains and subdirectories.
```
Comment (phallobst) [inline API.md]:
```suggestion
If you're using our API to match client URLs with our dataset make sure you only use the domain of the `url`-element, as a website commonly uses subdomains and subdirectories.
```

**Review by phallobst — CHANGES_REQUESTED** [review]

**Comment by Carlgo11** [comment]
I put v1 at the top to preserve chronological order but I guess we could use reverse chronological order. Should the remarks still be at the bottom then?

**Comment by phallobst** [comment]
Having an introduction (along with some development notes/remarks) at the top seems to be better, I think.

**Comment by Carlgo11** [comment]
@phallobst I've done a couple of drafts with your suggestions but they all suffer in the readability aspect.
As the remarks rely on Version 2 having been read it can only really be below Version 2. That means that the order would be Version 2, Remarks, Version 1 which looks really weird.

So basically, how strong do you feel about the changes? Because making the changes would require rewriting the entire document in order for it to be coherent.

**Comment by phallobst** [comment]
What about his one? I didn't really change that much, and it's still quite readable (although lacking a strong story arc).

----

# API usage

## Introduction

The data collected for the twofactorauth.org website is also available as JSON files in order to enable developers to use it in their own programs. It is recommended to use the API with the highest version number, since older versions might not include all available information.

### URL & Domain matching

If you're using our API to match client URLs with our dataset make sure you only use the domain of the `url`-element, as a website commonly uses subdomains and subdirectories. Please note that there are exceptions like `.co.uk`, `.com.au` or `.co.nz` where the actual domain is found at a lower level.

### Caching

If you intent to query our JSON files often and with a lot of traffic, you may be blocked by Cloudflare, our reverse proxy provider. We therefore recommend that you cache the files locally for any large traffic cases.

### Avoid downloading unnecessary data

If you only intent on using a specific dataset, like all sites supporting RFC-6238, we recommend that you use the URI which lists just that. See [URIs](#uris-1) for a list of available paths. The smaller the better.

## Version 2
[...]

## Version 1 (Deprecated) :warning:
[...]

**Review by phallobst — APPROVED** [review]

**Review by dsoegijono — APPROVED** [review]

**Pull request merged**